src/main/java/de/retest/recheck/review/ignore/ColorDiffFilter.java

```diff
@@ -0,0 +1,125 @@
+package de.retest.recheck.review.ignore;
+
+import static java.lang.Float.parseFloat;
+import static java.lang.Math.abs;
+import static java.lang.Math.min;
+import static java.lang.Math.round;
+
+import java.awt.Color;
+import java.io.Serializable;
+import java.util.Optional;
+import java.util.regex.MatchResult;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import de.retest.recheck.ignore.Filter;
+import de.retest.recheck.review.ignore.io.RegexLoader;
+import de.retest.recheck.ui.descriptors.Element;
+import de.retest.recheck.ui.diff.AttributeDifference;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+@Getter
+@Slf4j
+public class ColorDiffFilter implements Filter {
+
+	private static final Pattern RGB = Pattern.compile( "rgb\\((\\d+),\\s?(\\d+),\\s?(\\d+)(,\\s?(\\d+))?\\)" );
+	private static final double MAX_DISTANCE = 255;
+
+	private final String givenInput;
+	private final double colorDiff;
+
+	public ColorDiffFilter( final String givenInput, final double colorDiff ) {
+		this.givenInput = givenInput;
+		this.colorDiff = colorDiff;
+	}
+
+	@Override
+	public boolean matches( final Element element ) {
+		return false;
+	}
+
+	@Override
+	public boolean matches( final Element element, final AttributeDifference attributeDifference ) {
+		final Color expected = parse( attributeDifference.getExpected() );
+		final Color actual = parse( attributeDifference.getActual() );
+
+		if ( expected == null || actual == null ) {
+			return false;
+		}
+
+		final double distance = calculateColorDistance( expected, actual ) * 100;
+
+		return distance < colorDiff;
+	}
+
+	/**
+	 * Return the color distance as a value in the range of [0.0, 1.0]. Note this is about perception: any one of the
+	 * rgba values is reviewed in isolation, as a change from e.g. white to red or green to yellow already constitutes a
+	 * 100% change in the perceived color.
+	 */
+	static double calculateColorDistance( final Color expected, final Color actual ) {
+		final double r1 = expected.getRed();
```

Comment:
It may not look nice in L71, but can you use some more descriptive name schema like `redActual`/`redExpected` and so on?

Reply:
Done